### PR TITLE
Major overhaul to ConsoleLogger

### PR DESCRIPTION
**Pull request by gwynne**
`ConsoleLogger` (and some other stuff) has been overhauled as follows:

- Additional ANSI escape codes have been added
- The latest `swift-log` is now required
- `ConsoleLogger` can now be configured in considerably more detail, such as control over which log message elements appear at what log levels and what console styles are used for each individual element.
- A new `logging-demo` command is available in the `ConsoleKitExample`.
- On macOS, `input(isSecure: true)` now uses `readpassphrase()` instead of `getpass()`. A Windows implementation using `_getch()` has been added. An improved Linux implementation is pending a separate PR.

### CODE REVIEW

File: Sources/ConsoleKit/Logging/ConsoleLogger.swift [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+    /// The output configuration which applies to the logger backed by this handler. Initially, a copy of the default
+    /// configuration provided when boostrapping the logging system, but can be updated to change the output behavior of
+    /// the individual `Logger`.
+    public var outputConfiguration: Logger.OutputConfiguration = .default
+
+    /// The style configuration which applies to the logger backed by this handler. Initially, a copy of the default
+    /// configuration provided when boostrapping the logging system, but can be updated to change the style of the
+    /// individual `Logger`.
+    public var styleConfiguration: Logger.StyleConfiguration = .default
```
Comment (tanner0101) [inline Sources/ConsoleKit/Logging/ConsoleLogger.swift:27]: What about this structure instead:

```swift
struct Configuration {
    struct Output { ... }
    struct Style { ... }
}
```

File: Sources/ConsoleKit/Logging/ConsoleLogger.swift [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+        var text: ConsoleText = ""
+        
+        // Note: The use of `" ".consoleText(self.styleConfiguration.baseStyle)` and similar constructs seems pointless at
+        //       first glance, but its necessity becomes immediately obvious after, for example, giving each of the styles
+        //       a different (and non-default) background color.
+        // TODO: Don't torment the construction of the consoleText pieces so much. This can be done ENORMOUSLY more efficiently.
+
+        // Render the timestamp if currently visible.
+        if self.outputConfiguration.timestampDisplayLogLevel.map({ $0 >= self.logLevel }) ?? false {
+            text += "[".consoleText(self.styleConfiguration.baseStyle) +
```
Comment (tanner0101) [inline Sources/ConsoleKit/Logging/ConsoleLogger.swift:89]: I wonder if the string interpolating API would make this easier to read? For example:
```swift
"[ \("INFO", style: .info) ]"
```

File: Sources/ConsoleKit/Logging/LoggerStyleConfiguration.swift [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        
+        /// A dictionary mapping log levels to styles. If an entry exists for a given level, that style _replaces_ the
+        /// base style for log messages with that level. The base style is used for all other levels.
+        let levelStyles: [Logger.Level: ConsoleStyle]
+        
+        /// The style used for a log message's timestamp.
+        let timestampStyle: ConsoleStyle?
+        
+        /// The style used for a log message's label, when visible.
+        let labelStyle: ConsoleStyle?
```
Comment (tanner0101) [inline Sources/ConsoleKit/Logging/LoggerStyleConfiguration.swift:21]: `styleConfiguration.labelStyle`. We probably don't need `Style` twice? `logger.configuration.style.timestamp = .info`

File: Sources/ConsoleKit/Terminal/Terminal.swift [hunk 4]
```diff
@@ -43,12 +45,35 @@ public final class Terminal: Console {
     public func input(isSecure: Bool) -> String {
         didOutputLines(count: 1)
         if isSecure {
-            // http://stackoverflow.com/a/30878869/2611971
+#if os(macOS)
+            var buffer = Array<Int8>.init(repeating: 0, count: 1024)
+            while readpassphrase("", &buffer, buffer.count, RPP_REQUIRE_TTY) == nil {
```
Comment (tanner0101) [inline Sources/ConsoleKit/Terminal/Terminal.swift:50]: +1 to this. Can we have it as a separate PR?

**Review by Moddyblue — APPROVED** [review]

**Pull request closed without merging**